File: leetcode/Array/max_profit.cpp
#include <bits/stdc++.h>

using namespace std;

int maxProfit(const vector<int>& prices) {
	// brute force solution
	int maxProfit = 0;
	for (int i = 0; i < prices.size() - 1; i++) {
		for (int j = 1 ; j < prices.size() - i; j++) {
			maxProfit = max(maxProfit, prices[i + j] - prices[i]);
		}
	}
	return maxProfit;
}

int main() {
	vector<int> arr = {1, 3, 1, 4, 5, 9, 2, 8, 1};
	cout << maxProfit(arr) << endl;
}

File: leetcode/Array/max_profit.rs
use std::cmp::max;

fn maxProfit(prices: Vec<i32>) -> i32 {
    let mut maxProfit: i32 = 0;
    for i in 0..prices.len() - 1 {
        for j in 1..prices.len() - i {
            maxProfit = max(maxProfit, prices[i + j] - prices[i]);
        }
    }
    return maxProfit;
}

fn main() {
    println!("max profit: {}", maxProfit(vec![1, 3, 1, 4, 5, 9, 2, 8, 1]));
}
